manager/controllers/app/policy_compiler_interface.go

```diff
@@ -6,75 +6,58 @@ package app
+	}
+	action.SetProcessingLocation(operation.GetDestination())
+	req.SetAction(action)
+
+	reqContext := make(map[string]interface{})
+	for k, v := range input.Spec.AppInfo {
+		reqContext[k] = v
+	}
+	req.SetContext(reqContext)
+
 	var credentialPath string
 	if input.Spec.SecretRef != "" {
 		credentialPath = utils.GetVaultAddress() + vault.PathForReadingKubeSecret(input.Namespace, input.Spec.SecretRef)
 	}
-	return &pb.ApplicationContext{
-		AppInfo: &pb.ApplicationDetails{
-			ProcessingGeography: operation.Destination,
-			Properties:          input.Spec.AppInfo,
-		},
-		CredentialPath: credentialPath,
-		Datasets: []*pb.DatasetContext{{
-			Dataset: &pb.DatasetIdentifier{
-				DatasetId: datasetID,
-			},
-			Operation: operation,
-		}},
-	}
+
+	return &req, credentialPath, nil
 }
 
 // LookupPolicyDecisions provides a list of governance actions for the given dataset and the given operation
-func LookupPolicyDecisions(datasetID string, policyManager connectors.PolicyManager, input *app.FybrikApplication, op *pb.AccessOperation) ([]*pb.EnforcementAction, error) {
+func LookupPolicyDecisions(datasetID string, policyManager connectors.PolicyManager, input *app.FybrikApplication, op *openapiclientmodels.PolicyManagerRequestAction) ([]*openapiclientmodels.ResultItem, error) {
 	// call external policy manager to get governance instructions for this operation
-	appContext := ConstructApplicationContext(datasetID, input, op)
-	openapiReq, creds, _ := connectors.ConvertGrpcReqToOpenAPIReq(appContext)
-	log.Println("transformed openapi request: ", openapiReq)
+	openapiReq, creds, _ := ConstructOpenAPIReq(datasetID, input, op)
+	log.Println("constructred openapi request: ", openapiReq)
 	openapiResp, err := policyManager.GetPoliciesDecisions(openapiReq, creds)
-	log.Println("openapi response received: ", openapiResp)
-	pcresponse, _ := connectors.ConvertOpenAPIRespToGrpcResp(openapiResp, datasetID, op)
-	log.Println("transformed grpc response: ", pcresponse)
-
-	actions := []*pb.EnforcementAction{}
-	if err != nil {
-		return actions, err
-	}
-
-	for _, datasetDecision := range pcresponse.GetDatasetDecisions() {
-		if datasetDecision.GetDataset().GetDatasetId() != datasetID {
-			continue // not our data set
-		}
-		operationDecisions := datasetDecision.GetDecisions()
-		for _, operationDecision := range operationDecisions {
-			enforcementActions := operationDecision.GetEnforcementActions()
-			for _, action := range enforcementActions {
-				if utils.IsDenied(action.GetName()) {
-					var message string
-					switch operationDecision.Operation.Type {
-					case pb.AccessOperation_READ:
-						message = app.ReadAccessDenied
-					case pb.AccessOperation_WRITE:
-						message = app.WriteNotAllowed
-					}
-					return actions, errors.New(message)
-				}
-				// Check if this is a real action (i.e. not Allow)
-				if utils.IsAction(action.GetName()) {
-					actions = append(actions, action)
-				}
-			}
-		}
```

Comment:
Why has this code been removed? 
1. In case of Deny it returns a corresponding error message
2. Allow actions are not appended

Not sure about Allow, perhaps it is not being returned at all, but indicating an error as a result of governance decision is essential.

Reply:
Thanks @shlomitk1 . I have now changed the relevant part. The unit tests are passing. However, we may still need some changes to remove some hardcoding of action names and ids (in the new taxonomy definitions there is nothing like action ID only action name is provided ) in the code. This is similar to the comment (https://github.com/fybrik/fybrik/pull/808#discussion_r708532214)  in #808 . @ritwikc @roee88 .

Reply:
@rohithdv  Are we expecting Allow actions to be a thing in the new version?

Reply:
Right now in #808 we have a rule which configures "Deny by default" (https://github.com/ritwikc/the-mesh-for-data/blob/2d010cd398af5bc88fd6f9d8dbd974b948283e64/third_party/opa/data-and-policies/user-created-policy-3/sample_policies.rego#L5). If that is the case then whenever this rule is loaded, there is no question of Allow. Either it is Redact or Deny.

Reply:
The idea following previous discussions on "by default" was that it's allow unless otherwise specified. A user that wants to switch to deny by default would add the rule as in our example. 

We can have the "by default" policy loaded at installation time directly so there won't be a risk of synchronization delay.

Reply:
just to clarify, we should have an "Allow" action defined in the base taxonomy which is used by default ? Also should the hardcoding in the code be changed?

Reply:
I was thinking that there will not be an allow action. If there are no actions in the result and 
there is no error, then it's interpreted as allow.

Reply:
Updated [evaluate_opa.go](https://github.com/rohithdv/fybrik/blob/e7f8d19d2e18bdde0e99e55de19c2529cd2a7555/connectors/open_policy_agent/lib/evaluate_opa.go#L91) and [mockup policymanager.go](https://github.com/rohithdv/fybrik/blob/e7f8d19d2e18bdde0e99e55de19c2529cd2a7555/manager/controllers/mockup/policymanager.go#L39) to return empty result in case of Allow. Please check. Thanks. @roee88